Issue/12934 add header to p2 postcard to show author

khaykov opened the pull request:
Fixes #12934

This PR modifies a header of the P2 reader posts to show the author's avatar alongside the blog avatar and show the author's name alongside the blog's name.
Where the p2 has not blavatar only the author's avatar will be shown (or no avatar at all if neither are available).

| Before | After Dark | After Light |
|--------|--------|--------|
| [![Image from Gyazo](https://i.gyazo.com/56a9ffafa4d9717bf3e3ddf067a9c898.png)](https://gyazo.com/56a9ffafa4d9717bf3e3ddf067a9c898) |[![Image from Gyazo](https://i.gyazo.com/900e6c9adc83c6cabfcb90d3f9c91e8a.png)](https://gyazo.com/900e6c9adc83c6cabfcb90d3f9c91e8a)|[![Image from Gyazo](https://i.gyazo.com/f8e2694a58729421cb121badf7b5d55f.png)](https://gyazo.com/f8e2694a58729421cb121badf7b5d55f)|

To test:
- To test DB migration install this branch over any previous version (eg. develop). Older posts already stored in DB without P2 flag will not show P2 specific layout.
- Make sure you follow a p2 site.
- Navigate to the reader and check that the p2 site's postcard header show's author's avatar next to the blog avatar.
- Make sure that the author's name is shown next to the blog name.
- Make sure those changes are not visible on non-p2 posts or in post details (including details of P2 post).

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

khaykov committed:
Added p2 flag to reader post.

khaykov committed:
Updated blog section UI state.

khaykov committed:
Updated UI state builder and post list view holder to show p2-specific layout.

khaykov committed:
Added p2 blog name with author resource.

khaykov committed:
Hiding author's avatar in post details.

khaykov committed:
Updated tests.

khaykov committed:
Fixed reader version comment.

khaykov committed:
Fixed lint errors, cleanup.

khaykov committed:
Removed test string.

khaykov committed:
Updated layout and colors.

khaykov committed:
Updated naming + comment.

khaykov committed:
RTL p2 site name support.

mbshakti commented:
This looks awesome @khaykov! Thank you for working on this. Apologies for my delay in reviewing this PR. Just one question, is there a reason we want to only show the first name of the author? (instead of the full name). Otherwise, everything looks great to me.

khaykov commented:
@mbshakti I used the logic from x-post - when there is no first name, the full name will be shown. I assume the original intent was to show the first name to save some space for the site title.

develric reviewed (commented):
Hey @khaykov 👋 . Code and feature looks really good, good job 👍 ! Tested on OnePlus 7T with Android 10. Also checked the db migration works well.

One thing I noticed is that we do not place the avatar image in front for x-post. if I'm not wrong we got inspired to place it in front looking on how it currently looks on iOS for x-post so probably we should align that also (if you agree I let you decide if here or in another PR).

This said I'm fine with merging this one, unless Shakti feels different about the author name topic 🙇 !

khaykov committed:
Put author above blavatar for xpost.

khaykov commented:
@develric thanks for the review! I agree about the avatar position in xpost. It's a pretty simple change, so I added it to this PR. Also changed the avatar BG to be the same as a postcard.

develric reviewed (approved):
Thanks @khaykov LGTM 👍 ! I'm moving this PR on since we need it as base for other modifications we need to do. @mbshakti in case you think there is anything we overlooked or just we need to clarify, please just let us know and we can create a dedicated PR addressing it if needed, thanks 🙇
